client/lib/domains/email-forwarding/index.js

```diff
@@ -1,14 +1,16 @@
 import emailValidator from 'email-validator';
 import { mapValues } from 'lodash';
 
-function validateAllFields( fieldValues ) {
+function validateAllFields( fieldValues, existingForwards = [] ) {
 	return mapValues( fieldValues, ( value, fieldName ) => {
 		const isValid = validateField( {
 			value,
 			name: fieldName,
 		} );
 
-		return isValid ? [] : [ 'Invalid' ];
+		return isValid
+			? validateDuplicatedForward( { value, name: fieldName }, existingForwards )
+			: [ 'Invalid' ];
```

Comment:
Not mandatory at all but here is another way to implement that logic that I find a bit easier to read:

```suggestion
		if ( ! isValid ) {
			return [ 'Invalid' ];
		}
		
		return validateDuplicatedForward( { value, name: fieldName }, existingForwards );
```